src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp

```diff
@@ -329,31 +329,39 @@ void prepare_primitive_fusing::fuse_bias(program &p) {
-            auto new_mem = new_node.get_attached_memory_ptr();
-            if (original_mem->count() != new_mem->count() || original_mem->get_layout().data_type != new_mem->get_layout().data_type)
+            auto second_mem = second_node.get_attached_memory_ptr();
+            if (original_mem->count() != second_mem->count() || original_mem->get_layout().data_type != second_mem->get_layout().data_type)
                 return false;
 
             switch (original_mem->get_layout().data_type) {
                 case data_types::f32: {
-                    mem_lock<float, mem_lock_type::write> original_bias_mem(original_mem, p.get_stream());
-                    mem_lock<float, mem_lock_type::read> new_bias_mem(new_mem, p.get_stream());
+                    cldnn::memory_ptr new_mem = p.get_engine().allocate_memory(original_mem->get_layout());
+                    mem_lock<float, mem_lock_type::write> new_bias_mem(new_mem, p.get_stream());
+                    mem_lock<float, mem_lock_type::read> original_bias_mem(original_mem, p.get_stream());
+                    mem_lock<float, mem_lock_type::read> second_bias_mem(second_mem, p.get_stream());
                     float* original_data = original_bias_mem.data();
-                    float* new_data = new_bias_mem.data();
+                    float* new_data = second_bias_mem.data();
                     for (size_t i = 0; i < original_bias_mem.size(); i++)
-                        original_data[i] += new_data[i];
+                        new_bias_mem[i] = original_data[i] + new_data[i];
+
+                    original_node.attach_memory(new_mem);
                     break;
                 }
                 case data_types::f16: {
-                    mem_lock<uint16_t, mem_lock_type::write> original_bias_mem(original_mem, p.get_stream());
-                    mem_lock<uint16_t, mem_lock_type::read> new_bias_mem(new_mem, p.get_stream());
+                    cldnn::memory_ptr new_mem = p.get_engine().allocate_memory(original_mem->get_layout());
+                    mem_lock<uint16_t, mem_lock_type::write> new_bias_mem(new_mem, p.get_stream());
+                    mem_lock<uint16_t, mem_lock_type::read> original_bias_mem(original_mem, p.get_stream());
+                    mem_lock<uint16_t, mem_lock_type::read> second_bias_mem(second_mem, p.get_stream());
                     uint16_t* original_data = original_bias_mem.data();
-                    uint16_t* new_data = new_bias_mem.data();
+                    uint16_t* new_data = second_bias_mem.data();
                     for (size_t i = 0; i < original_bias_mem.size(); i++) {
                         float new_val = half_to_float(original_data[i]) + half_to_float(new_data[i]);
-                        original_data[i] = float_to_half(new_val);
+                        new_bias_mem[i] = float_to_half(new_val);
                     }
+
+                    original_node.attach_memory(new_mem);
```

Comment:
why do you replace original_data? is it also related to igpu optimization?

Reply:
I faced a problem in fusion tests for this pass, because both (fused_ and ref_) networks use the same data memory, and during program build of fused_network this memory is modified by this optimization pass which affects ref_network's memory as well and leads to incorrect results
Also, I think it would be more correctly, since in real scenario during Constant Op creation different constants may be reused by several primitives (if constant values are the same), and then modification of data w/o taking 
into account it's users may lead to data invalidation for other users after this optimization. This change allows to keep original memory unchanged and attaches new one

Reply:
oh I see.. thanks for the explanation.